codegen/table.go

```diff
@@ -90,5 +92,50 @@ func NewTableFromStruct(name string, obj interface{}, opts ...TableOption) (*Tab
 		}
 	}
 
-	return t, nil
+	// add PK options
+	columns := make(ColumnDefinitions, 0, len(t.Columns))
+	for _, column := range t.Columns {
+		if _, ok := t.extraPKColumns[column.Name]; ok {
+			column.Options.PrimaryKey = true
+		}
+		columns = append(columns, column)
+	}
+	t.Columns = columns
+
+	return t, t.Check()
+}
+
+// Check that the resulting TableDefinition is correct (e.g., no overlapping column names).
+func (t *TableDefinition) Check() error {
+	if t == nil {
+		return fmt.Errorf("nil table definition")
+	}
+
+	if len(t.Name) == 0 {
+		return fmt.Errorf("empty table name")
+	}
+
+	if len(t.Columns) == 0 {
+		return fmt.Errorf("no columns for table %s", t.Name)
+	}
+
+	hasPK := false
+	columns := make(map[string]bool, len(t.Columns))
+	for _, column := range t.Columns {
+		switch {
+		case column.Type == schema.TypeInvalid:
+			return fmt.Errorf("%s->%s: invalid column type", t.Name, column.Name)
+		case columns[column.Name]:
+			return fmt.Errorf("%s->%s: duplicate column name", t.Name, column.Name)
+		default:
+			columns[column.Name] = true
+			hasPK = hasPK || column.Options.PrimaryKey
+		}
+	}
+
+	if !hasPK {
```

Comment:
Is this always an error? I think we can have tables without PKs and then we use a random ID via `_cq_id`.

Maybe we can return different errors so the caller can check and allow for some (for example X table can not have a primary key)

Reply:
Dropping this check for now